core/iwasm/common/wasm_shared_memory.c

```diff
@@ -417,8 +417,14 @@ wasm_runtime_atomic_wait(WASMModuleInstanceCommon *module, void *address,
 
     os_mutex_lock(&wait_info->wait_list_lock);
 
-    if ((!wait64 && *(uint32 *)address != (uint32)expect)
-        || (wait64 && *(uint64 *)address != expect)) {
+    WASMSharedMemNode *node =
```

Comment:
Could you declare `node` and `no_wait` below at the beginning of the function? We usually keep this C coding style to avoid compile warning for some compilers.